packages/react-google-maps-api/src/useLoadScript.tsx

```diff
@@ -16,6 +17,17 @@ export interface UseLoadScriptOptions extends LoadScriptUrlOptions {
 
 let previouslyLoadedUrl: string
 
+const useIsMounted = () => {
+  const isMountedRef = React.useRef(false)
+  React.useEffect(function trackMountedState() {
+    isMountedRef.current = true
+    return () => {
+      isMountedRef.current = false
+    }
+  }, [])
+  return useCallback(() => isMountedRef.current,[]);
```

Comment:
Why callback? It does need referential integrity. I think it's better to just return a whole ref.

Reply:
And I would probably move this whole hook into a separate file. I can imagine it will find its use in other places eventually.

Reply:
About ref, I personnally prefer returning a isMounted() function, which seems to be simpler to manipulate as it's just a js concept, not a react one). Don't remember who said that but "use the least powerful abstraction that gets the job done". I guess you meant referential transparency, because integrity seems related to RDBMS for me.

Will extract that in a separate file